app/models/user.rb

```diff
@@ -37,10 +39,38 @@ def github_client
   end
 
   def github_client_scopes
-    GitHubUser.new(github_client, uid).client_scopes
+    GitHub::Token.scopes(token, github_client)
   end
 
   def staff?
     site_admin
   end
+
+  private
+
+  # Internal: We need to make sure that the user
+  # doesn't reduce the scope of their token. In
+  # the event that their token is needed to
+  # perform other functions.
+  #
+  # If the token that is trying to be set is
+  # lower than the one we have toss it, and use the one we have.
+  #
+  # If the token has the same scopes but is newer, or has more scopes
+  # let the token be set.
+  #
+  # See https://github.com/education/classroom/issues/445
+  def ensure_no_token_scope_loss
+    return true if token_was.blank?
+    return true unless token_changed?
+
+    old_scopes = GitHub::Token.scopes(token_was)
+    new_scopes = GitHub::Token.scopes(token)
+
+    # This is naive, if the token scopes ever change
+    # come back an revist this.
+    return true if old_scopes.size < new_scopes.size
```

Comment:
don't love this, but it would only matter if we change the scopes we request for the different roles